docs/design_docs/011-historacle-pricing.md

```diff
@@ -0,0 +1,88 @@
+# Design Doc 011: Historacle Pricing
+
+## Status
+
+Draft
+
+## Abstract
+
+In order to support small-volume assets in a safe manner, we need to be able to calculate the weighted medians of our prices over the past X amount of time (30 days or so). This will allow the leverage module to make decisions around when to allow liquidations if an asset price is considered abnormal.
+
+## Context
+
+Right now, the volume of assets such as Juno, Umee, and Osmo are so low that it would be relatively easy to use the leverage protocol on UMEE to manipulate the prices of these assets and game the system.
+
+The attack goes:
+
+1. Account A deposits & collateralizes a large amount of USDC.
+2. Account A uses collateral to borrow a lot of the low-volume token (here, FOO).
+3. Account A sells the tokens to Account B, inflating the price of FOO.
+4. Account B collateralizes inflated FOO, and borrows USDC.
+5. Account B exits the market for a profit.
+
+In order to avoid this, and to continue with our goal of allowing users to collateralize low-volume assets, we need to have a safety net.
```

Comment:
```suggestion
In order to avoid these attacks, and to continue with our goal of allowing users to collateralize and borrow low-volume assets, we need to have a safety net.
```
`these attacks` only if mentioning both scenarios